DIP/DIP/Models/CardPayment.cs

```diff
@@ -0,0 +1,28 @@
+﻿using DIP.Interfaces;
+
+namespace DIP
+{
+    public class CardPayment : IPayment
+    {
+        public int Total { get; set; }
+        public double Discount { get; set; }
+        public CardPayment(int total, int discount)
+        {
+            Total = total;
+            Discount = discount/100;
+        }
+        public CardPayment()
+        {
+        }
+        public string Pay()
+        {
+            if (Discount < 1 && 0 < Discount) {
+                double totalAfterDiscount = Total - (Total * Discount);
+                Console.WriteLine($"Payment Successful. Total: {Total}, After Discount {totalAfterDiscount}");
```

Comment:
this should not be in this class, also, we don't need this to be logged to the console. just return it in the respose